brian2tools/plotting/synapses.py

```diff
@@ -182,7 +182,10 @@ def plot_synapses(sources, targets, values=None, var_unit=None,
             origin = kwds.pop('origin', 'lower')
             interpolation = kwds.pop('interpolation', 'nearest')
             axes.imshow(matrix, origin=origin, interpolation=interpolation,
-                        cmap=cmap, norm=norm, **kwds)
+                        cmap=cmap, norm=norm,
+                        extent=(min(unique_sources), max(unique_sources),
+                                min(unique_targets), max(unique_targets)),
+                        **kwds)
```

Comment:
I think if you want the pixels to be centred on the correct values, you need to do something like subtract 0.5 from the min values and add 0.5 to the max values, otherwise the centres won't be consistent.